src/NzbDrone.Core.Test/DecisionEngineTests/UpgradeAllowedSpecificationFixture.cs

```diff
@@ -0,0 +1,293 @@
+                new QualityModel(Quality.FLAC),
+                Language.English
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_same_quality_when_upgrading_is_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English),
+                    Cutoff = Language.English,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_320),
+                Language.English
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_same_quality_when_upgrading_is_not_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = false
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English),
+                    Cutoff = Language.English,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_320),
+                Language.English
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_lower_quality_when_upgrading_is_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English),
+                    Cutoff = Language.English,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_256),
+                Language.English
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_lower_quality_when_upgrading_is_not_allowed()
```

Comment:
I'd expect `false` given lower quality and upgrade not allowed

Reply:
Yea, I would agree on this one...